src/addons/Portal/Portal.js

```diff
@@ -229,6 +273,12 @@ class Portal extends Component {
     this.trySetState({ open: false })
   }
 
+  closeWithTimeout = (e, delay) => {
+    // React wipes certain props (e.g. currentTarget) so we need to clone.
+    const eventClone = { ...e }
```

Comment:
Heads up, React wipes the entire event object, nullifying all values.  That's because React reuses a single event object.  In order to persist the event object for async access, use `e.persist()`:

https://facebook.github.io/react/docs/events.html#event-pooling

I think what is here will technically work, but it still seems as though we should be using `e.persist()`, even if we copying the persisted object.